src/main/java/it/gov/pagopa/pu/debtpositions/service/create/debtposition/CreateDebtPositionServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+                                       InstallmentNoPIIRepository installmentNoPIIRepository) {
+    this.authorizeOperatorOnDebtPositionTypeService = authorizeOperatorOnDebtPositionTypeService;
+    this.debtPositionMapper = debtPositionMapper;
+    this.validateDebtPositionService = validateDebtPositionService;
+    this.debtPositionService = debtPositionService;
+    this.generateIuvService = generateIuvService;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+  }
+
+  @Override
+  public DebtPositionDTO createDebtPosition(DebtPositionDTO debtPositionDTO, Boolean massive, Boolean pagopaPayment) {
+    log.info("START Create DebtPosition...");
+    String accessToken = SecurityUtils.getAccessToken();
+
+    verifyAuthorization(debtPositionDTO);
+    validateInstallments(debtPositionDTO, accessToken);
+
+    DebtPosition debtPosition = debtPositionMapper.mapToModel(debtPositionDTO).getFirst();
+
+    debtPosition.getPaymentOptions().stream()
+      .flatMap(po -> po.getInstallments().stream())
+      .forEach(installmentNoPII -> {
+        long countDuplicates = installmentNoPIIRepository.countExistingDebtPosition(debtPosition.getOrganizationId(), installmentNoPII.getIud(), installmentNoPII.getIuv(), installmentNoPII.getNav());
+        if (countDuplicates > 0) {
+          log.error("Duplicate record found for Installment with id {}", installmentNoPII.getInstallmentId());
+          throw new ConflictErrorException("Duplicate records found: the provided data conflicts with existing records.");
+        }
+      });
+
+    if (Boolean.TRUE.equals(pagopaPayment)) {
+      debtPositionDTO.getPaymentOptions().stream()
+        .flatMap(po -> po.getInstallments().stream())
+        .forEach(installment -> {
+          String orgFiscalCode = installment.getDebtor().getFiscalCode();
+          String generatedIuv = generateIuvService.generateIuv(orgFiscalCode, accessToken);
+          installment.iuv(generatedIuv);
+        });
+    }
+
+    DebtPositionDTO savedDebtPosition = debtPositionService.saveDebtPosition(debtPositionDTO);
+
+    log.info("END Create DebtPosition...");
+    return savedDebtPosition;
+  }
+
+  private void verifyAuthorization(DebtPositionDTO debtPositionDTO) {
+    log.info("START Verify Authorization...");
+
+    String operatorExternalUserId = SecurityUtils.getCurrentUserExternalId();
+    DebtPositionTypeOrg debtPositionTypeOrg = authorizeOperatorOnDebtPositionTypeService.authorize(debtPositionDTO.getOrganizationId(), debtPositionDTO.getDebtPositionTypeOrgId(), operatorExternalUserId);
+
+    log.info("END Verify Authorization, found DebtPositionTypeOrg id: {}", debtPositionTypeOrg.getDebtPositionTypeOrgId());
+  }
+
+  private void validateInstallments(DebtPositionDTO debtPositionDTO, String accessToken) {
+    log.info("START Validating installments...");
+
+    validateDebtPositionService.validate(debtPositionDTO, accessToken);
+
+    log.info("END validating installments");
+  }
```

Comment:
too many logs, I will not configure them and just call directly the service on the public method, same for the method  verifyAuthorization